From 12ccdc3c0c61c7c9de013d3208e9627ac7529c78 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 6 Nov 2018 14:59:18 +0100
Subject: [PATCH 6/6] [SQUSH][DNM] Merge tag 'android-9.0.0_r16' into
 lineage-16.0

commit 874887072a9eaf6977b35eae91ac6140dfc090df
Merge: ab2e9a27f 273af6882
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Sep 10 23:21:26 2018 +0000

    Merge cherrypicks of [4986743, 4986744, 4987539, 4987168, 4986376, 4986377, 4986378, 4986889, 4986745, 4986746, 4986747, 4986748, 4986749, 4986750, 4986773, 4987169, 4987170, 4987611, 4987631, 4987632, 4987633, 4987634, 4986890, 4987612, 4987651, 4987598, 4987613, 4987614, 4987615, 4987599, 4986379, 4986380, 4987652, 4987653, 4987691, 4986774] into pi-qpr1-release

    Change-Id: I181deec3875449ab991738f9e831e6e46834ab1a

commit 273af68825de1ea2d1e4038088c45f80376e2e43
Author: Santhosh Behara <santhoshbehara@codeaurora.org>
Date:   Thu Aug 16 17:46:15 2018 +0530

    mm-video-v4l2: vdec: Correct order of reading color primaries

    According to H265 spec, the color primaries should be in G,B,R
    order. But in the exisisting code they are read in R, G, B order.
    Correct the order of reading bitstream as per the specification.

    (cherry-picked from:
    https://partner-android-review.googlesource.com/c/platform/hardware/qcom/media/+/1123749/)
    Bug: 111799726
    Bug: 112443029
    Change-Id: Ic6703dcf7878c27e3e18e437039ddb74cf8d32d6
    (cherry picked from commit defe2fa2ca6553b686de3cbac7f65d51584e16fe)
    (cherry picked from commit a9ce4757f75ad260ba6c022cd875ba21fb3f8089)

commit ab2e9a27fed1550c95c9a8afba547e4848c9d38a
Merge: 4e8f57444 259f413df
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 09:16:25 2018 -0700

    [automerger skipped] Merge changes from topic "am-afec395a-dee0-4bb8-a6d1-e1a546a9e5cc" into oc-dev am: 27c62d3683  -s ours am: f3724f0c53  -s ours
    am: 259f413df4  -s ours

    Change-Id: Iae52e2d1e2270f93ddef39b8cc94f6664e8d6e9d

commit 4e8f57444ecbabb361a410a3a6837ee31a69a278
Merge: 2af4d2861 47b968798
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 09:16:10 2018 -0700

    [automerger skipped] Merge changes from topic "am-eb7e5809-a8f1-4883-95d8-8a81d24349c2" into oc-dev am: 86cd0cefbb  -s ours am: 66795bb553  -s ours
    am: 47b9687981  -s ours

    Change-Id: I9ac1e5b3612402e3ea7b35c8e0ad6881ef0e1bac

commit 259f413df43bf94dbf05a7192667bcbd168c1c24
Merge: 47b968798 f3724f0c5
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 09:12:23 2018 -0700

    [automerger skipped] Merge changes from topic "am-afec395a-dee0-4bb8-a6d1-e1a546a9e5cc" into oc-dev am: 27c62d3683  -s ours
    am: f3724f0c53  -s ours

    Change-Id: I6dc7d58eb454756febb5e5d062d6a383e2b9f2fa

commit 47b96879817ff0199cff1874dfa9a20fac95fa58
Merge: 144423242 66795bb55
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 09:12:10 2018 -0700

    [automerger skipped] Merge changes from topic "am-eb7e5809-a8f1-4883-95d8-8a81d24349c2" into oc-dev am: 86cd0cefbb  -s ours
    am: 66795bb553  -s ours

    Change-Id: I2cfa800f93f01d703c7cd274bc95d426b8bd7e4a

commit f3724f0c537adee9eee00a1bca13612690f1a12a
Merge: 66795bb55 27c62d368
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 09:08:23 2018 -0700

    [automerger skipped] Merge changes from topic "am-afec395a-dee0-4bb8-a6d1-e1a546a9e5cc" into oc-dev
    am: 27c62d3683  -s ours

    Change-Id: I6a8ba10c34708979c45f721df719771cc0613f6b

commit 66795bb553dffd41b6af93875cb363cae0baed39
Merge: 64e207938 86cd0cefb
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 09:08:09 2018 -0700

    [automerger skipped] Merge changes from topic "am-eb7e5809-a8f1-4883-95d8-8a81d24349c2" into oc-dev
    am: 86cd0cefbb  -s ours

    Change-Id: I879b1c21f5a92cd2171f8666e2059e29f9306006

commit 27c62d368337d4a6425abb84781e121c5de3183e
Merge: 86cd0cefb 4dee36e0b
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 16:03:42 2018 +0000

    Merge changes from topic "am-afec395a-dee0-4bb8-a6d1-e1a546a9e5cc" into oc-dev

    * changes:
      [automerger skipped] mm-video-v4l2: Squash below changes skipped: 93850b4d18
      mm-video-v4l2: Squash below changes

commit 86cd0cefbb80d8e352b94f18d9e451d26705d2b6
Merge: c1a44d640 2e024c80a
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 16:03:34 2018 +0000

    Merge changes from topic "am-eb7e5809-a8f1-4883-95d8-8a81d24349c2" into oc-dev

    * changes:
      [automerger skipped] mm-video-v4l2: Protect buffer access and increase input buffer size skipped: c3897ad4e4 skipped: 61dfbccfa2 skipped: 652ed92486 skipped: 8ee5c834b4 skipped: b9c4d4e766
      [automerger skipped] mm-video-v4l2: Protect buffer access and increase input buffer size skipped: c3897ad4e4 skipped: 61dfbccfa2 skipped: 652ed92486 skipped: 8ee5c834b4
      [automerger skipped] mm-video-v4l2: Protect buffer access and increase input buffer size skipped: c3897ad4e4 skipped: 61dfbccfa2 skipped: 652ed92486
      [automerger skipped] mm-video-v4l2: Protect buffer access and increase input buffer size skipped: c3897ad4e4 skipped: 61dfbccfa2
      [automerger skipped] mm-video-v4l2: Protect buffer access and increase input buffer size skipped: c3897ad4e4
      mm-video-v4l2: Protect buffer access and increase input buffer size

commit 2af4d28614442c0de3845ae39096bf264b7fc56b
Merge: ab0f8d44e 144423242
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 06:43:05 2018 -0700

    [automerger skipped] Merge changes from topic "am-ce370b5f-d000-493e-89f3-ac566d0f1112" into oc-dev am: c1a44d6404  -s ours am: 64e207938a  -s ours
    am: 1444232423  -s ours

    Change-Id: Ib00bc2b1935357ca73c32cd1a54768f5ec161ff8

commit ab0f8d44e2031ae5f039557f6db80b5e5885a9ab
Merge: 9f488f84c bc84a5db0
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 06:42:50 2018 -0700

    [automerger skipped] Merge "mm-video-v4l2: Squash below changes" into oc-dev am: c78dd4781c  -s ours am: 1ab0d4f0d3  -s ours
    am: bc84a5db0f  -s ours

    Change-Id: Ifcf527ba5b12844f6729b95223f1abd89cb07bd0

commit 14442324239ec0c411d983a8ba17f31aed398e46
Merge: bc84a5db0 64e207938
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 06:39:05 2018 -0700

    [automerger skipped] Merge changes from topic "am-ce370b5f-d000-493e-89f3-ac566d0f1112" into oc-dev am: c1a44d6404  -s ours
    am: 64e207938a  -s ours

    Change-Id: I70da3cad1c819a212f6ec90174b28bd8b4e00dfb

commit bc84a5db0f938dd479652352c4646505b7fe57b9
Merge: 232c22ffd 1ab0d4f0d
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 06:38:51 2018 -0700

    [automerger skipped] Merge "mm-video-v4l2: Squash below changes" into oc-dev am: c78dd4781c  -s ours
    am: 1ab0d4f0d3  -s ours

    Change-Id: Id2affac79c14d56a3e56bb45c803821a5427564e

commit 64e207938afe05a56b286cdbe1df8b8b146da603
Merge: 1ab0d4f0d c1a44d640
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 06:35:05 2018 -0700

    [automerger skipped] Merge changes from topic "am-ce370b5f-d000-493e-89f3-ac566d0f1112" into oc-dev
    am: c1a44d6404  -s ours

    Change-Id: I792879074d0105369de0038097349fcf6be6dd36

commit 1ab0d4f0d34d6af01ef7ccd50c7908863b4823b2
Merge: 7f60d7094 c78dd4781
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 06:34:51 2018 -0700

    [automerger skipped] Merge "mm-video-v4l2: Squash below changes" into oc-dev
    am: c78dd4781c  -s ours

    Change-Id: I37d41a28b5c5791def3db80fb7a430252fce9db0

commit c1a44d6404e108033babfb2aea4b412b15a83f3e
Merge: c78dd4781 978ceb55f
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 13:32:30 2018 +0000

    Merge changes from topic "am-ce370b5f-d000-493e-89f3-ac566d0f1112" into oc-dev

    * changes:
      [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 6b19d10e03 am: 9ddc8509a8 am: e26b05c060 skipped: 0fbb8162de
      [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 6b19d10e03 am: 9ddc8509a8 am: e26b05c060
      [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 6b19d10e03 am: 9ddc8509a8
      [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 6b19d10e03
      mm-video-v4l2: Protect buffer access and increase input buffer size

commit c78dd4781c6078d8ae7cba27b961c69b117f83b0
Merge: 1747c75df cdbc75cd8
Author: Ray Essick <essick@google.com>
Date:   Sat Jul 14 13:31:51 2018 +0000

    Merge "mm-video-v4l2: Squash below changes" into oc-dev

commit 9f488f84c593cccacd00094d3696e53ef582017e
Merge: 9382c4615 232c22ffd
Author: Ray Essick <essick@google.com>
Date:   Fri Jul 13 21:36:51 2018 -0700

    [automerger skipped] Merge "mm-video-v4l2: Protect buffer access and increase input buffer size" into oc-dev am: 1747c75df1 am: 7f60d70943
    am: 232c22ffd5  -s ours

    Change-Id: I1b1a9a9741119693f74eebd7dc06d8ed2a7872ff

commit 232c22ffd5c83b55e5cf07f165be938a6d2174ab
Merge: 39d6e9a90 7f60d7094
Author: Ray Essick <essick@google.com>
Date:   Fri Jul 13 21:25:17 2018 -0700

    Merge "mm-video-v4l2: Protect buffer access and increase input buffer size" into oc-dev am: 1747c75df1
    am: 7f60d70943

    Change-Id: I80a7f3d0d3909d820dbaca312927e0a1eae77847

commit 9382c46158d10d751943a988157787b3c1187293
Merge: fa68facb1 39d6e9a90
Author: Ray Essick <essick@google.com>
Date:   Fri Jul 13 21:25:15 2018 -0700

    [automerger skipped] Merge "mm-video-v4l2: Squash below changes" into oc-dr1-dev am: d5a9640fef
    am: 39d6e9a909  -s ours

    Change-Id: Iaa2347be6b217b63c35f002ef992c56ee5714cbb

commit 7f60d7094352047a3f322d56013e4534fb84e2b2
Merge: d5a9640fe 1747c75df
Author: Ray Essick <essick@google.com>
Date:   Fri Jul 13 21:23:15 2018 -0700

    Merge "mm-video-v4l2: Protect buffer access and increase input buffer size" into oc-dev
    am: 1747c75df1

    Change-Id: I218434126085e5a2ea26302758a4afc5fa73d847

commit 39d6e9a909005496bbd40a5514fc3d55d9f660b7
Merge: 48dd91c89 d5a9640fe
Author: Ray Essick <essick@google.com>
Date:   Fri Jul 13 21:22:16 2018 -0700

    Merge "mm-video-v4l2: Squash below changes" into oc-dr1-dev
    am: d5a9640fef

    Change-Id: I42e0e9d9fd51f5b3ca6a2b7bedf079291f618a8e

commit 1747c75df1ece8d487f792345f1fc02fa262108d
Merge: 7314778d7 660db9e97
Author: TreeHugger Robot <treehugger-gerrit@google.com>
Date:   Sat Jul 14 00:29:25 2018 +0000

    Merge "mm-video-v4l2: Protect buffer access and increase input buffer size" into oc-dev

commit d5a9640fef0722f3c83b10c0c13d1136a2f8fda0
Merge: bfc24b86a 808b94b16
Author: TreeHugger Robot <treehugger-gerrit@google.com>
Date:   Sat Jul 14 00:28:35 2018 +0000

    Merge "mm-video-v4l2: Squash below changes" into oc-dr1-dev

commit fa68facb1cc1b4dca6e46e4c19ff940d0ab7b172
Merge: 150fcf6e2 48dd91c89
Author: Santhosh Behara <santhoshbehara@codeaurora.org>
Date:   Fri Jul 13 17:19:35 2018 -0700

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 83aeab22d1 am: 2a0384e026 am: c68a7ce355 am: 32e9465138 am: 04f8e526b4 am: 7314778d75 am: bfc24b86ac
    am: 48dd91c89c

    Change-Id: I393d1d913c4851e9adfbbc28dc73a25e59037342

commit 48dd91c89cf8d07177f16b6a0332f00ad2f10817
Merge: 9db527e44 bfc24b86a
Author: Santhosh Behara <santhoshbehara@codeaurora.org>
Date:   Fri Jul 13 17:06:17 2018 -0700

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 83aeab22d1 am: 2a0384e026 am: c68a7ce355 am: 32e9465138 am: 04f8e526b4 am: 7314778d75
    am: bfc24b86ac

    Change-Id: I18fd25029704ab2e7b5a164f22ce43c5b224bfc4

commit bfc24b86ac2b19ee29649f504b2e4664153f82b6
Merge: 9d54deadf 7314778d7
Author: Santhosh Behara <santhoshbehara@codeaurora.org>
Date:   Fri Jul 13 15:43:02 2018 -0700

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 83aeab22d1 am: 2a0384e026 am: c68a7ce355 am: 32e9465138 am: 04f8e526b4
    am: 7314778d75

    Change-Id: Iefff1c8402c68b7aa2c23d95f5b6724e6e399ef4

commit 150fcf6e2136d00f9859e2a7b5b2be75e35b781b
Merge: ff8cc2b25 5e811c42a
Author: TreeHugger Robot <treehugger-gerrit@google.com>
Date:   Fri Jul 13 20:53:48 2018 +0000

    Merge "mm-video-v4l2: Protect buffer access and increase input buffer size" into pi-dev

commit 978ceb55f00064025f1e54919705d74047513394
Merge: f6bc54126 0fbb8162d
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Tue Jul 3 17:12:12 2018 +0000

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 6b19d10e03 am: 9ddc8509a8 am: e26b05c060 skipped: 0fbb8162de

    Change-Id: I05af9eeb40737c9bf85fa6dc91e4931d06b91450

commit 2e024c80ab8b94fa05ad9f9e8e2459fea76f7ab2
Merge: f6bc54126 b9c4d4e76
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Mon Jul 2 23:11:03 2018 +0000

    [automerger skipped] mm-video-v4l2: Protect buffer access and increase input buffer size skipped: c3897ad4e4 skipped: 61dfbccfa2 skipped: 652ed92486 skipped: 8ee5c834b4 skipped: b9c4d4e766

    Change-Id: Ia47e91c44d0706bcbf0216a5cb40c2705f73f737

commit b9c4d4e7661179f5e26ee047c3747701ea016bb5
Merge: edce745b3 8ee5c834b
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Mon Jul 2 23:11:02 2018 +0000

    [automerger skipped] mm-video-v4l2: Protect buffer access and increase input buffer size skipped: c3897ad4e4 skipped: 61dfbccfa2 skipped: 652ed92486 skipped: 8ee5c834b4

    Change-Id: I806d22d6caf7c485a075335cc096ef16ec036e1b

commit 8ee5c834b4a76465446761a9b26651b8edc83c6f
Merge: e8c693ae3 652ed9248
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Mon Jul 2 23:11:01 2018 +0000

    [automerger skipped] mm-video-v4l2: Protect buffer access and increase input buffer size skipped: c3897ad4e4 skipped: 61dfbccfa2 skipped: 652ed92486

    Change-Id: I5cb7af90c27fff6001b4391e993992859a4c548a

commit 652ed92486d1fab71dff8c4b5f36785350671e3d
Merge: 74e9f7581 61dfbccfa
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Mon Jul 2 23:11:00 2018 +0000

    [automerger skipped] mm-video-v4l2: Protect buffer access and increase input buffer size skipped: c3897ad4e4 skipped: 61dfbccfa2

    Change-Id: I6da84bbd54d04277bd1b097f3f98ff482ea67ed9

commit 61dfbccfa2ee25cef41a5ce1ff64c77f00e419cd
Merge: 5e19db925 c3897ad4e
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Mon Jul 2 23:10:58 2018 +0000

    [automerger skipped] mm-video-v4l2: Protect buffer access and increase input buffer size skipped: c3897ad4e4

    Change-Id: I9c20f9048ea5697a394311841c2e4d1dfcab598f

commit c3897ad4e4a49a41e7c7a3589f41cf5c662b165f
Author: Ray Essick <essick@google.com>
Date:   Mon Jul 2 16:09:33 2018 -0700

    mm-video-v4l2: Protect buffer access and increase input buffer size

    Protect buffer access for below scenarios:

    *Increase the scope of buf_lock in free_buffer to avoid access
     of freed buffer for both input and output buffers. Also, add check
     before output buffer access.

    *Disallow allocate buffer mode after client has called use buffer.

    Allocate additional 512 bytes of memory for input buffers on top of
    allocation size as per hardware requirement.

    Bug: 64340487
    Test: ran POC on sailfish/nyc-mr1-dev
    Change-Id: I441d7f33af557f384f13b1f50d6c808153e9f37a
    Merged-In: I93cb2da613a4adc01bf0529a677e270bf4956fe6
    Merged-In: Ic13b511b2d1c46e0b158880c41c9a02e2fefd113
    Merged-In: Id699404d6c276c8702826d79b3a6ba9755b3439f

commit 0fbb8162de29664b392d3cb226227aaecfc56985
Merge: edce745b3 e26b05c06
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Mon Jul 2 23:06:40 2018 +0000

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 6b19d10e03 am: 9ddc8509a8 am: e26b05c060

    Change-Id: I1670a912bda5e6a4caceaf2d2dc0500a66bd5f18

commit e26b05c0600fa8de5f78f22eeca92b29b4a0d059
Merge: e8c693ae3 9ddc8509a
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Mon Jul 2 23:06:39 2018 +0000

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 6b19d10e03 am: 9ddc8509a8

    Change-Id: I7bd73aea1cfc662ee4772a486514b1ac27f7cc1b

commit 9ddc8509a8d0879fadac282f9e58b7988b9326ea
Merge: 74e9f7581 6b19d10e0
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Mon Jul 2 23:06:36 2018 +0000

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 6b19d10e03

    Change-Id: I6bbb60e993fc8ca3b705f1eddbfe8f69da660209

commit 6b19d10e030f0ed087cad5cea17fd3a296bc25be
Author: Ray Essick <essick@google.com>
Date:   Mon Jul 2 16:05:54 2018 -0700

    mm-video-v4l2: Protect buffer access and increase input buffer size

    Protect buffer access for below scenarios:

    *Increase the scope of buf_lock in free_buffer to avoid access
     of freed buffer for both input and output buffers. Also, add check
     before output buffer access.

    *Disallow allocate buffer mode after client has called use buffer.

    Allocate additional 512 bytes of memory for input buffers on top of
    allocation size as per hardware requirement.

    Bug: 64340487
    Test: ran POC on sailfish/nyc-mr1-dev
    Change-Id: I93cb2da613a4adc01bf0529a677e270bf4956fe6
    Merged-In: Ic13b511b2d1c46e0b158880c41c9a02e2fefd113
    Merged-In: Id699404d6c276c8702826d79b3a6ba9755b3439f

commit 660db9e9708872f9a407ac14e92a5017599ced34
Author: Ray Essick <essick@google.com>
Date:   Mon Jul 2 16:00:09 2018 -0700

    mm-video-v4l2: Protect buffer access and increase input buffer size

    Protect buffer access for below scenarios:

    *Increase the scope of buf_lock in free_buffer to avoid access
     of freed buffer for both input and output buffers. Also, add check
     before output buffer access.

    *Disallow allocate buffer mode after client has called use buffer.

    Allocate additional 512 bytes of memory for input buffers on top of
    allocation size as per hardware requirement.

    Bug: 64340487
    Test: ran POC on sailfish/nyc-mr1-dev
    Change-Id: Ic13b511b2d1c46e0b158880c41c9a02e2fefd113
    Merged-In: Id699404d6c276c8702826d79b3a6ba9755b3439f

commit 5e811c42a3c29f77c95ebf2eaa8975c5c2058ca3
Author: Ray Essick <essick@google.com>
Date:   Mon Jul 2 15:57:21 2018 -0700

    mm-video-v4l2: Protect buffer access and increase input buffer size

    Protect buffer access for below scenarios:

    *Increase the scope of buf_lock in free_buffer to avoid access
     of freed buffer for both input and output buffers. Also, add check
     before output buffer access.

    *Disallow allocate buffer mode after client has called use buffer.

    Allocate additional 512 bytes of memory for input buffers on top of
    allocation size as per hardware requirement.

    Bug: 64340487
    Test: ran POC on sailfish/nyc-mr1-dev
    Change-Id: Id699404d6c276c8702826d79b3a6ba9755b3439f

commit 4dee36e0b005d492c60ee61a874c432748892ce4
Merge: f6bc54126 93850b4d1
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Fri Jun 29 23:22:39 2018 +0000

    [automerger skipped] mm-video-v4l2: Squash below changes skipped: 93850b4d18

    Change-Id: Ibbf92921383b516a1cb853eb54c58817707f5a83

commit 93850b4d18a54837c0b6b2e71e104f49cc99377e
Author: Ray Essick <essick@google.com>
Date:   Fri Jun 29 16:20:00 2018 -0700

    mm-video-v4l2: Squash below changes

    mm-video-v4l2: Protect buffer access and increase input buffer size

    Protect buffer access for below scenarios:
    *Increase the scope of buf_lock in free_buffer to avoid access
     of freed buffer for both input and output buffers. Also, add check
     before output buffer access.
    *Disallow allocate buffer mode after client has called use buffer.
    Allocate additional 512 bytes of memory for input buffers on top of
    allocation size as per hardware requirement.

    mm-video-v4l2: correct the buffer error flag setting

    The buffer error flag setting is corrected in free_buffer()
    call

    Bug: 64340487
    Test: poc from bug
    Change-Id: I7c3a58dc6ed7401e8b7cf626adbea6801e8801a7
    Merged-In: I33c26e1a8d7f60825358441c7fde28d2f04adc02
    Merged-In: I8a28d86135f2fc39a45a6a07355845327dec2dfa
    Merged-In: I32d6cb7f61b873edd567881d1bf3e620cd78e715

commit cdbc75cd863590b89ed3e45c14eb2324e34cf4b1
Author: Ray Essick <essick@google.com>
Date:   Fri Jun 29 16:12:11 2018 -0700

    mm-video-v4l2: Squash below changes

    mm-video-v4l2: Protect buffer access and increase input buffer size

    Protect buffer access for below scenarios:
    *Increase the scope of buf_lock in free_buffer to avoid access
     of freed buffer for both input and output buffers. Also, add check
     before output buffer access.
    *Disallow allocate buffer mode after client has called use buffer.
    Allocate additional 512 bytes of memory for input buffers on top of
    allocation size as per hardware requirement.

    mm-video-v4l2: correct the buffer error flag setting

    The buffer error flag setting is corrected in free_buffer()
    call

    Bug: 64340487
    Test: poc from bug
    Change-Id: I33c26e1a8d7f60825358441c7fde28d2f04adc02
    Merged-In: I8a28d86135f2fc39a45a6a07355845327dec2dfa
    Merged-In: I32d6cb7f61b873edd567881d1bf3e620cd78e715

commit 808b94b164c39d75d78481f301151fc299b201cc
Author: Ray Essick <essick@google.com>
Date:   Fri Jun 29 14:12:19 2018 -0700

    mm-video-v4l2: Squash below changes

    mm-video-v4l2: Protect buffer access and increase input buffer size

    Protect buffer access for below scenarios:
    *Increase the scope of buf_lock in free_buffer to avoid access
     of freed buffer for both input and output buffers. Also, add check
     before output buffer access.
    *Disallow allocate buffer mode after client has called use buffer.
    Allocate additional 512 bytes of memory for input buffers on top of
    allocation size as per hardware requirement.

    mm-video-v4l2: correct the buffer error flag setting

    The buffer error flag setting is corrected in free_buffer()
    call

    Bug: 64340487
    Test: poc from bug
    Change-Id: I8a28d86135f2fc39a45a6a07355845327dec2dfa
    Merged-In: I32d6cb7f61b873edd567881d1bf3e620cd78e715

commit ff8cc2b25beca2401208e5e04840f58b8f4055fd
Author: Ray Essick <essick@google.com>
Date:   Fri Jun 29 11:22:06 2018 -0700

    mm-video-v4l2: Squash below changes

    mm-video-v4l2: Protect buffer access and increase input buffer size

    Protect buffer access for below scenarios:
    *Increase the scope of buf_lock in free_buffer to avoid access
     of freed buffer for both input and output buffers. Also, add check
     before output buffer access.
    *Disallow allocate buffer mode after client has called use buffer.
    Allocate additional 512 bytes of memory for input buffers on top of
    allocation size as per hardware requirement.

    mm-video-v4l2: correct the buffer error flag setting

    The buffer error flag setting is corrected in free_buffer()
    call

    Bug: 64340487
    Test: poc from bug
    Change-Id: I32d6cb7f61b873edd567881d1bf3e620cd78e715

commit 7314778d75cbdc1adb4e2d7b1932170074e16f47
Merge: f6bc54126 04f8e526b
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Fri Jun 22 17:02:42 2018 +0000

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 83aeab22d1 am: 2a0384e026 am: c68a7ce355 am: 32e9465138 am: 04f8e526b4

    Change-Id: I1e64eea0d3bf5e03b8a041e22efbd2b476a53469

commit 04f8e526b4fa1e8f05df5421232b97c9c0e88730
Merge: edce745b3 32e946513
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Fri Jun 22 17:02:41 2018 +0000

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 83aeab22d1 am: 2a0384e026 am: c68a7ce355 am: 32e9465138

    Change-Id: Icecfb38eb3d808ed4215abe3db1bcc04befc8276

commit 32e9465138092316dc6265f227815e6e8fe784f0
Merge: e8c693ae3 c68a7ce35
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Fri Jun 22 17:02:39 2018 +0000

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 83aeab22d1 am: 2a0384e026 am: c68a7ce355

    Change-Id: I4a39cb279fa905270d61192d7fc832bd3c7d9fa7

commit c68a7ce35523de7b82272f027f2e4c573a1e83d0
Merge: 74e9f7581 2a0384e02
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Fri Jun 22 17:02:37 2018 +0000

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 83aeab22d1 am: 2a0384e026

    Change-Id: I50c491c2301cc8382f40251da2cf6f7e1ed13c45

commit 2a0384e026b7c6b4cf99e85950f0bb587b15f23f
Merge: 5e19db925 83aeab22d
Author: Android Build Merger (Role) <noreply-android-build-merger@google.com>
Date:   Fri Jun 22 17:02:36 2018 +0000

    [automerger] mm-video-v4l2: Protect buffer access and increase input buffer size am: 83aeab22d1

    Change-Id: I4e7ad1ed9de14f79b802d7091371d22e1d88762c

commit 83aeab22d1bdc493b3ea2f50616bb8fd460d6c74
Author: Santhosh Behara <santhoshbehara@codeaurora.org>
Date:   Tue May 15 06:09:50 2018 -0700

    mm-video-v4l2: Protect buffer access and increase input buffer size

    Protect buffer access for below scenarios:

    *Increase the scope of buf_lock in free_buffer to avoid access
     of freed buffer for both input and output buffers. Also, add check
     before output buffer access.

    *Disallow allocate buffer mode after client has called use buffer.

    Allocate additional 512 bytes of memory for input buffers on top of
    allocation size as per hardware requirement.

    Bug: 64340487
    Test: ran POC on bullhead/nyc-dev
    Change-Id: Iabbb2d7e00ff97bfc47b04386feec66976fca99a

commit edce745b365445c6dd4df01a941a369bdb8cd00e
Merge: d1d8a3d2e a8c07e024
Author: Ray Essick <essick@google.com>
Date:   Fri May 4 17:52:10 2018 +0000

    Merge changes from topic "am-de30185d-edfc-49d5-b5e1-aa1bdb83d97a" into nyc-mr2-dev

    * changes:
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db am: 7240438e1f am: 7f08f16258 am: 2fd221ebe8 am: e071682ab1
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db am: 7240438e1f am: 7f08f16258 am: 2fd221ebe8
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db am: 7240438e1f am: 7f08f16258
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db am: 7240438e1f
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504
      Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"

commit e8c693ae37715035347d1ff28b91c64b740467ce
Merge: 56ad27d28 e071682ab
Author: Ray Essick <essick@google.com>
Date:   Fri May 4 17:52:10 2018 +0000

    Merge changes from topic "am-de30185d-edfc-49d5-b5e1-aa1bdb83d97a" into cw-f-dev

    * changes:
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db am: 7240438e1f am: 7f08f16258 am: 2fd221ebe8
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db am: 7240438e1f am: 7f08f16258
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db am: 7240438e1f
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504
      Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"

commit 74e9f7581e69ec93aa488f353238837060bf42f9
Merge: 160015e71 2fd221ebe
Author: Ray Essick <essick@google.com>
Date:   Fri May 4 17:52:10 2018 +0000

    Merge changes from topic "am-de30185d-edfc-49d5-b5e1-aa1bdb83d97a" into nyc-mr1-dev

    * changes:
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db am: 7240438e1f am: 7f08f16258
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db am: 7240438e1f
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504
      Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"

commit 5e19db92575f23056680c17f52fd5cb14b50ff52
Merge: a7a02e6e2 7f08f1625
Author: Ray Essick <essick@google.com>
Date:   Fri May 4 17:52:10 2018 +0000

    Merge changes from topic "am-de30185d-edfc-49d5-b5e1-aa1bdb83d97a" into nyc-dr1-dev

    * changes:
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db am: 7240438e1f
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504
      Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"

commit 503672691dfbc3d54c6feb08b6ea6698e9ea2949
Merge: b77266aed 7240438e1
Author: Ray Essick <essick@google.com>
Date:   Fri May 4 17:52:10 2018 +0000

    Merge changes from topic "am-de30185d-edfc-49d5-b5e1-aa1bdb83d97a" into nyc-dev

    * changes:
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1 skipped: d4481934db
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d am: d43877ace1
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504 am: f8f1a5f63d
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: e1dfe8f504
      Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"

commit d1d8a3d2e02cb59ed6da63e70def614b4af28d6e
Merge: 6ca5bc05e 229feaed1
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri May 4 17:50:07 2018 +0000

    Merge changes from topic "am-96a5eb3e-2317-4e97-b0f7-e4891ab64c97" into nyc-mr2-dev

    * changes:
      [automerger skipped] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" skipped: 2e2293b82e skipped: 5d8ba5e98f skipped: 76637d6380 skipped: 525005edf5
      [automerger skipped] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" skipped: 2e2293b82e skipped: 5d8ba5e98f skipped: 76637d6380
      [automerger skipped] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" skipped: 2e2293b82e skipped: 5d8ba5e98f
      [automerger skipped] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" skipped: 2e2293b82e
      Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"

commit 56ad27d28dc7a129d642918152fe568416777b42
Merge: f990c7a4c 525005edf
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri May 4 17:50:07 2018 +0000

    Merge changes from topic "am-96a5eb3e-2317-4e97-b0f7-e4891ab64c97" into cw-f-dev

    * changes:
      [automerger skipped] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" skipped: 2e2293b82e skipped: 5d8ba5e98f skipped: 76637d6380
      [automerger skipped] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" skipped: 2e2293b82e skipped: 5d8ba5e98f
      [automerger skipped] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" skipped: 2e2293b82e
      Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"

commit 160015e7142fbe8d7d83143d9dd22e7bbd05a300
Merge: 188d112d9 76637d638
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri May 4 17:50:07 2018 +0000

    Merge changes from topic "am-96a5eb3e-2317-4e97-b0f7-e4891ab64c97" into nyc-mr1-dev

    * changes:
      [automerger skipped] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" skipped: 2e2293b82e skipped: 5d8ba5e98f
      [automerger skipped] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" skipped: 2e2293b82e
      Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"

commit a7a02e6e218ab3418f3c12196fe1da41ada68064
Merge: a37cae442 5d8ba5e98
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri May 4 17:50:07 2018 +0000

    Merge changes from topic "am-96a5eb3e-2317-4e97-b0f7-e4891ab64c97" into nyc-dr1-dev

    * changes:
      [automerger skipped] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" skipped: 2e2293b82e
      Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"

commit b77266aeddb75a70b141299daedd33175a738c0b
Merge: 41923ca7b 2e2293b82
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri May 4 17:50:07 2018 +0000

    Merge "Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"" into nyc-dev

commit 6ca5bc05eae32a98019f271ec24d46e19d5d51d8
Merge: 53f90b755 7f917145d
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri May 4 17:34:48 2018 +0000

    Merge changes from topic "am-23973cb8-6ed2-4797-9749-019d6251e3c6" into nyc-mr2-dev

    * changes:
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: b2255366c0 am: 188d112d9c am: f990c7a4c9
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: b2255366c0 am: 188d112d9c
      [automerger] Revert "mm-video-v4l2: Protect buffer access and increase input buffer size" am: b2255366c0
      Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"

commit a37cae442fbeca26624740ece90ea230d7dbf22c
Merge: 2c4e77cd2 b2255366c
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri May 4 17:34:48 2018 +0000

    Merge "Revert "mm-video-v4l2: Protect buffer access and increase input buffer size"" into nyc-dr1-dev

Change-Id: I787403a5a5fa640ba5ef5d271b6dc69f25494496
---
 .../vidc/vdec/src/omx_vdec_v4l2.cpp           | 24 +++++++++----------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/msm8998/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp b/msm8998/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp
index b61f49b3c..419d9f5f7 100644
--- a/msm8998/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp
+++ b/msm8998/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp
@@ -11399,12 +11399,12 @@ bool omx_vdec::handle_mastering_display_color_info(void* data, MasteringDisplay*
     mastering_display_mdata->maxDisplayLuminance = mastering_display_payload->nMaxDisplayMasteringLuminance;
     mastering_display_mdata->minDisplayLuminance = mastering_display_payload->nMinDisplayMasteringLuminance;
 
-    internal_disp_changed_flag |= (hdr_info->sType1.mR.x != mastering_display_payload->nDisplayPrimariesX[0]) ||
-        (hdr_info->sType1.mR.y != mastering_display_payload->nDisplayPrimariesY[0]);
-    internal_disp_changed_flag |= (hdr_info->sType1.mG.x != mastering_display_payload->nDisplayPrimariesX[1]) ||
-        (hdr_info->sType1.mG.y != mastering_display_payload->nDisplayPrimariesY[1]);
-    internal_disp_changed_flag |= (hdr_info->sType1.mB.x != mastering_display_payload->nDisplayPrimariesX[2]) ||
-        (hdr_info->sType1.mB.y != mastering_display_payload->nDisplayPrimariesY[2]);
+    internal_disp_changed_flag |= (hdr_info->sType1.mG.x != mastering_display_payload->nDisplayPrimariesX[0]) ||
+        (hdr_info->sType1.mG.y != mastering_display_payload->nDisplayPrimariesY[0]);
+    internal_disp_changed_flag |= (hdr_info->sType1.mB.x != mastering_display_payload->nDisplayPrimariesX[1]) ||
+        (hdr_info->sType1.mB.y != mastering_display_payload->nDisplayPrimariesY[1]);
+    internal_disp_changed_flag |= (hdr_info->sType1.mR.x != mastering_display_payload->nDisplayPrimariesX[2]) ||
+        (hdr_info->sType1.mR.y != mastering_display_payload->nDisplayPrimariesY[2]);
 
     internal_disp_changed_flag |= (hdr_info->sType1.mW.x != mastering_display_payload->nWhitePointX) ||
         (hdr_info->sType1.mW.y != mastering_display_payload->nWhitePointY);
@@ -11418,12 +11418,12 @@ bool omx_vdec::handle_mastering_display_color_info(void* data, MasteringDisplay*
         (hdr_info->sType1.mMinDisplayLuminance != mastering_display_payload->nMinDisplayMasteringLuminance);
 
     if (internal_disp_changed_flag) {
-        hdr_info->sType1.mR.x = mastering_display_payload->nDisplayPrimariesX[0];
-        hdr_info->sType1.mR.y = mastering_display_payload->nDisplayPrimariesY[0];
-        hdr_info->sType1.mG.x = mastering_display_payload->nDisplayPrimariesX[1];
-        hdr_info->sType1.mG.y = mastering_display_payload->nDisplayPrimariesY[1];
-        hdr_info->sType1.mB.x = mastering_display_payload->nDisplayPrimariesX[2];
-        hdr_info->sType1.mB.y = mastering_display_payload->nDisplayPrimariesY[2];
+        hdr_info->sType1.mG.x = mastering_display_payload->nDisplayPrimariesX[0];
+        hdr_info->sType1.mG.y = mastering_display_payload->nDisplayPrimariesY[0];
+        hdr_info->sType1.mB.x = mastering_display_payload->nDisplayPrimariesX[1];
+        hdr_info->sType1.mB.y = mastering_display_payload->nDisplayPrimariesY[1];
+        hdr_info->sType1.mR.x = mastering_display_payload->nDisplayPrimariesX[2];
+        hdr_info->sType1.mR.y = mastering_display_payload->nDisplayPrimariesY[2];
         hdr_info->sType1.mW.x = mastering_display_payload->nWhitePointX;
         hdr_info->sType1.mW.y = mastering_display_payload->nWhitePointY;
 
-- 
2.17.1

